source/class/cv/ui/structure/tile/Controller.js

```diff
@@ -0,0 +1,492 @@
+/**
+ * Controller for the 'tile' structure.
+ *
+ * This structure is based on web-components and does not need a parser to work.
+ * The config file is directly attached to the document body.
+ * The widgets in this structure register a customElement and the browser automatically creates instances
+ * of this widgets once the customElement is added to the body.
+ *
+ * The basic structure is a set of pages that contain a list of tiles.
+ * Each tile contains a grid of 3 rows and 3 columns. The components can be added to a cell of that grid
+ * but also can spread over more then one cell by using row-/column spanning.
+ *
+ * This structure provides some tiles with a pre-defined content, e.g. a <cv-switch> which
+ * contains of a button in the middle of the tile and a primary- and secondary label in the third row.
+ *
+ * Those pre-defined tiles are provided by a <template> (@see https://developer.mozilla.org/de/docs/Web/HTML/Element/template}
+ * User are able to define own templates for re-usable tiles if they need one that this structure does not provide.
+ *
+ * @asset(structures/tile/*)
+ * @author Tobias Bräutigam
+ * @since 2022
+ */
+qx.Class.define('cv.ui.structure.tile.Controller', {
+  extend: qx.core.Object,
+  type: 'singleton',
+  implement: cv.ui.structure.IController,
+
+  /*
+  ***********************************************
+    CONSTRUCTOR
+  ***********************************************
+  */
+  construct: function () {
+    this.base(arguments);
+    this.__HTML_STRUCT = '';
+    qx.bom.Stylesheet.includeFile(qx.util.ResourceManager.getInstance().toUri('designs/tile-globals.scss').replace('.scss', '.css') + (cv.Config.forceReload === true ? '?'+Date.now() : ''));
+  },
+
+  /*
+  ***********************************************
+    STATICS
+  ***********************************************
+  */
+  statics: {
+    // prefix for all custom components uses/provided by this structure
+    PREFIX: 'cv-',
+    __MAP: {},
+    __I: {},
+    register(webComponentName, qxClass) {
+      this.__MAP[webComponentName] = qxClass;
+    },
+
+    onComponentCreated(element) {
+      const name = element.tagName.toLowerCase();
+      if (Object.prototype.hasOwnProperty.call(this, name)) {
+        const QxClass = this.__MAP[name];
+        if (!Object.prototype.hasOwnProperty.call(this, name)) {
+          this.__I[name] = [];
+        }
+        this.__I[name].push(new QxClass(element));
+      } else {
+        qx.log.Logger.error('no QxClass registered for custom element ' + name);
```

Comment:
can't this be written as `this.error()` instead?

Reply:
No, `this.error` is only available in class instances, this is a static function where you have to use `qx.log.Logger.error(this, <error-message>` but I forgot the `this`, so I will fix that.